packages/plugin-ext/doc/how-to-add-new-plugin-namespace.md

```diff
@@ -1,22 +1,19 @@
-# This document describes how to add new plugin api namespace
+# How to add a new plugin API namespace
 
-New Plugin API namespace should be packaged as Theia extension
+This document describes how to add new plugin API namespace in the plugin host.
+Depending on the plugin host we can either provide a frontend or backend API extension:
 
-## Provide your API or namespace
+- In the backend plugin host that runs in the Node environment in a separate process, we adapt the module loading to return a custom API object instead of loading a module with a particular name.
+- In the frontend plugin host that runs in the browser environment via a web worker, we import the API scripts and put it in the global context.
 
-This API developed in the way that you provide your API as separate npm package.
-In that package you can declare your api.
-Example `foo.d.ts`:
+In this document we focus on the implementation of a backend plugin API.
+However, both APIs can be provided by implementing and binding an `ExtPluginApiProvider` which should be packaged as a Theia extension.
 
-```typescript
-    declare module '@bar/foo' {
-        export namespace fooBar {
-            export function getFoo(): Foo;
-        }
-    }
-```
+## Declare your plugin API provider
+
+The plugin API provider is executed on the respective plugin host to add your custom API namespace.
 
-## Declare `ExtPluginApiProvider` implementation
+Example Foo Plugin API provider:
```

Comment:
I recommend we name this `FooExtPluginApiProvider`. There is a `MainPluginApiProvider` that needs to be implemented as well.

Reply:
Good idea, I'll change it.